app/src/main/java/com/mapbox/maps/testapp/examples/annotation/SymbolClusterActivity.kt

```diff
@@ -0,0 +1,160 @@
+import kotlinx.android.synthetic.main.activity_annotation.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import java.util.*
+
+/**
+ * Example showing how to add Symbol cluster annotations
+ */
+class SymbolClusterActivity : AppCompatActivity(), CoroutineScope {
+  private var mapboxMap: MapboxMap? = null
+  private val job = Job()
+  override val coroutineContext = job + Dispatchers.IO
+  private var symbolManager: SymbolManager? = null
+  private var options: List<SymbolOptions>? = null
+  private var index: Int = 0
+  private val random = Random()
+  private val nextStyle: String
+    get() {
+      return AnnotationUtils.STYLES[index++ % AnnotationUtils.STYLES.size]
+    }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_annotation)
+    progress.visibility = View.VISIBLE
+    mapboxMap = mapView.getMapboxMap()
+      .apply {
+        jumpTo(
+          CameraOptions.Builder()
+            .center(Point.fromLngLat(LONGITUDE, LATITUDE))
+            .zoom(10.0)
+            .build()
+        )
+        loadStyleUri(nextStyle) {
+          val annotationPlugin = mapView.getAnnotationPlugin()
+          val annotationConfig = AnnotationConfig(
+            annotationSourceOptions = AnnotationSourceOptions(
+              clusterOptions = ClusterOptions(
+                textColorExpression = color(Color.YELLOW),
+                textColor = Color.BLACK, // Will not be applied as textColorExpression has been set
+                textSize = 20.0,
+                circleRadiusExpression = literal(25.0),
+                colorLevels = listOf(
+                  Pair(100, Color.RED),
+                  Pair(50, Color.BLUE),
+                  Pair(0, Color.GREEN)
+                )
+              )
+            )
+          )
+          symbolManager = annotationPlugin.createSymbolManager(mapView, annotationConfig)
+          launch {
+            loadData()
+          }
+        }
+      }
+
+    deleteAll.setOnClickListener { symbolManager?.deleteAll() }
+    changeStyle.setOnClickListener {
+      mapView.getMapboxMap().loadStyleUri(nextStyle)
+    }
+  }
+
+  private fun loadData() {
+    if (options == null) {
+      AnnotationUtils.loadStringFromNet(this@SymbolClusterActivity, POINTS_URL)?.let {
+        FeatureCollection.fromJson(it).features()?.let { features ->
+          features.shuffle()
+          options = features.take(AMOUNT).map { feature ->
+            SymbolOptions()
+              .withGeometry((feature.geometry() as Point))
+              .withIconImage(ICON_FIRE_STATION)
+          }
+        }
+      }
+    }
+    symbolManager?.deleteAll()
+    runOnUiThread {
+      options?.let {
+        symbolManager?.create(it)
+      }
+      progress.visibility = View.GONE
+    }
+  }
```

Comment:
Some of this code seems to be have optimized to be calling loadData more than once but if I'm only seeing loadData being called once? Can this be simplified?